processor/ratelimitprocessor/README.md

```diff
@@ -191,3 +193,113 @@ processors:
       window_multiplier: 1.5
       window_duration: 1m
 ```
+
+### Class-Based Dynamic Rate Limiting
+
+When using the Gubernator rate limiter type, you can define named rate limit classes with different base rates and escalation policies. A class resolver extension maps unique keys to class names, enabling different rate limits per customer tier, API version, or other business logic.
+
+Example with class-based configuration:
+
+```yaml
+extension:
+  configmapclassresolverextension:
+    name: resolutions
+    namespace: default
+
+processors:
+  ratelimiter:
+    metadata_keys:
+    - x-customer-id
+    rate: 100           # fallback rate
+    burst: 200          # fallback burst
+    type: gubernator
+    strategy: requests
+
+    # define the class resolver ID.
+    class_resolver: configmapclassresolverextension
+    # Define named classes
+    classes:
+      trial:
+        rate: 50        # 50 requests/second for trial users
+        burst: 100      # burst capacity of 100
+        static_only: true  # no dynamic escalation
+
+      paying:
+        rate: 500       # 500 requests/second for paying customers
+        burst: 1000     # burst capacity of 1000
+        static_only: false
+
+      enterprise:
+        rate: 2000      # 2000 requests/second for enterprise
+        burst: 4000     # burst capacity of 4000
+        static_only: false   # allow gradual increase.
+
+    # Default class when resolver returns unknown class
+    default_class: "trial"
```

Comment:
nit: wouldnt it be better to move this inside classes key to act as a catch all? 
```
classes:
   default: ? 

```

Reply:
That's a bit awkard since `classes` is a `map[string]Class`. That means that you always require a named class `"default"` to be declared, is that what you meant?